CHANGELOG.md

```diff
@@ -1,5 +1,8 @@
 # Release History
 
+## 1.2.0-dev.20210629 (Unreleased)
+
+
 ## 1.2.0-dev.20210628 (2021-06-28)
```

Comment:
I got worried for a second that this CL version had been checked-in.

Then noticed the target branch :)

Main still points to https://github.com/Azure/azure-sdk-for-c/blob/main/CHANGELOG.md#120-beta1-unreleased

cc @danieljurek